Make sure Q library doesn't suppress any exceptions (use done/nodeify after catches).

## corediff.spec.merge.js throws uncaught exceptions

e.g. https://github.com/webgme/webgme/blob/master/test/common/core/corediff.spec.merge.js#L884 results in:

    node ./node_modules/mocha/bin/mocha -R dot --recursive --timeout 10000 test/common/core/corediff.spec.merge.js

    unhandled error from callback { [AssertionError: expected '#fa3404d329b0be53648786400db65f301f09b501' to equal '#750269ed92ac349ed65d87ed4a91fe40381ea1a1']
      message: 'expected \'#fa3404d329b0be53648786400db65f301f09b501\' to equal \'#750269ed92ac349ed65d87ed4a91fe40381ea1a1\'',
      showDiff: true,
      actual: '#fa3404d329b0be53648786400db65f301f09b501',
      expected: '#750269ed92ac349ed65d87ed4a91fe40381ea1a1' }

    ...
      
    1) corediff-merge merge registry changing to the same value:
     Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.

I suggest using q to automatically catch these exceptions like e.g. https://github.com/webgme/webgme/blob/e55cf3a3cb20bb3553156dde6e7730f78d04344e/test/bin/usermanager.spec.js

--- a/test/bin/apply.spec.js
+++ b/test/bin/apply.spec.js
@@ -31,7 +31,7 @@ describe('apply CLI tests', function () {
                 return storage.openDatabase();
             })
             .then(function () {
-                testFixture.importProject(storage, {
+                return testFixture.importProject(storage, {
                     projectName: projectName,
                     logger: logger.fork('import'),
                     gmeConfig: gmeConfig,

--- a/test/bin/import.spec.js
+++ b/test/bin/import.spec.js
@@ -51,7 +51,7 @@ describe('import CLI tests', function () {
                 return storage.openDatabase();
             })
             .then(function () {
-                testFixture.importProject(storage, {
+                return testFixture.importProject(storage, {
                     projectName: existingProjectName,
                     logger: logger.fork('import'),
                     gmeConfig: gmeConfig,

--- a/test/common/core/setcore.spec.js
+++ b/test/common/core/setcore.spec.js
@@ -50,19 +50,15 @@ describe('set core', function () {
                 });
                 root = core.createNode();
             })
-            .then(done)
-            .catch(done);
+            .nodeify(done);
     });
 
     afterEach(function (done) {
         storage.deleteProject({projectId: projectId})
             .then(function () {
-                storage.closeDatabase(done);
+                return storage.closeDatabase();
             })
-            .catch(function (err) {
-                logger.error(err);
-                storage.closeDatabase(done);
-            });
+            .nodeify(done);
     });
 
     it('set members should be inherited as well', function () {

